cmake/CMakeDoxyfile.in

```diff
@@ -0,0 +1,280 @@
+#
+# DO NOT EDIT! THIS FILE WAS GENERATED BY CMAKE!
+#
```

Comment:
@majestrate Are these CMake files necessary? I'm unsure how i've generated so many and whether they should even be committed to this PR

Reply:
probably best to exclude them